www/SonectCordovaPlugin.js

```diff
@@ -3,10 +3,10 @@ var exec = require('cordova/exec');
 var payInternal = null;
 var checkBalanceInternal = null;
 
-exports.present = function(credentials, paymentMethods, theme, checkBalance, pay, success, error) {
+exports.present = function(credentials, paymentMethods, theme, paymentReference, checkBalance, pay, success, error) {
```

Comment:
Discussable part.

In Android in case of opening Activity there is no difference between first open of the SDK and process payment. Each time you need to provide credentials. From that point public API from Android part could be simple - just one method `present` which could have or have not payment reference.

In case of payment reference it will navigate directly to receipt screen.

Also could be useful if Neon wants to navigate from their UI (e.g. list of transaction) directly to particulart payment ref. They still need to provide credentials.


Or we could introduce new method.
Or it's still possible to adjust our SDK itself to store state in outer Plugin, but I'm not sure how Cordova handles plugins and if it will persist in memory for a while.

Reply:
It's the same on `iOS`. For that I use `presentTransaction`, which presents the SDK with the previous SDK intialization parameters, and passes the `paymentReference`. Can you use that instead of changing the public API?

Reply:
I could use that but still we need credentials there, right? I guess even in iOS.